### ISSUE TITLE
Weight loading

### ISSUE THREAD

**Issue by SeanBtmn** [LukasHedegaard/continual-inference#61]
Hi Lukas,

I've played around a bit with your framework and it looks great and speeds up my inference stage a lot.

But I'm not really sure what's the optimal way to train the model and then load the weights for inference.

What I've done so far:

- I have a Pytorch (non-continual) model with trained weights (all layers are supported by your framework and there is no padding on the time-dimension)
- I have (re-)implemented the model in your framework. 
- I realized I cant just load the weights from the Pytorch model into the continual model, since the weight names are different.
- I have manually mapped the weights from the Pytorch model to the weights of the continual model. This works now but it's really a big mess and it took me way too long.

Is there an easier way? How are you doing it? ;)

Best wishes,
Sean

**Comment by LukasHedegaard** [comment]
Hi Sean, 
Actually your approach isn't too different from what I usually do. If you need to train, I'd recommend you do it with the `forward` mode. 

When utilising weights trained elsewhere, the weight mapping can be a mess due the extra paths in the weight names, which the structured nesting adds. To alleviate this, I created a `flatten` option to the `state_dict` and `load_state_dict`, which will remove redundant path information from the weight name if possible. You can see an example of it [here](https://github.com/LukasHedegaard/continual-inference/blob/88fec55c4dbb4cef7bad25d9d7b902c07a651974/tests/continual/test_container.py#L373).

I hope it helps

**Comment by SeanBtmn** [comment]
It helps, thank you for the fast response!
